contracts/utils/Create2.sol

```diff
@@ -44,6 +44,11 @@ library Create2 {
         /// @solidity memory-safe-assembly
         assembly {
             addr := create2(amount, add(bytecode, 0x20), mload(bytecode), salt)
+            // if no address was created, and returndata is not empty, bubble revert
+            if and(iszero(addr), not(iszero(returndatasize()))) {
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
```

Comment:
This snippet is explicitly mentioned in Solidity documentation as something that is not memory-safe.

https://docs.soliditylang.org/en/latest/assembly.html#memory-safety

cc @Amxx @ernestognw

Reply:
See https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5057

Reply:
Been thinking about the consequences of this, but does it even matter given it's reverting? It's "memory unsafe" because it may overwrite memory but there's no way to do something unsafe afterwards.

Reply:
Yeah the docs say it matters if the assembly block reverts and this exact snippet is used to illustrate the point.

Reply:
<img width="739" alt="image" src="https://github.com/OpenZeppelin/openzeppelin-contracts/assets/481465/e5bbf38b-2e5a-4189-ae29-bd0bc4281e45">

Reply:
Yeah, it's kinda unclear what's the relationship between being "mainly about the optimizer" and still needing to ensure memory safe

Reply:
Yeah I should look into this for my next blog post to give you a reason to read it. 😁